src/geometry/WorkingGeometry.ts

```diff
@@ -130,4 +207,48 @@ export class WorkingGeometry {
             vertexCount += child.vertices.length;
         }
     }
+
+    /**
+     * Special divide for scaling vectors. Treats undeterminate as 0.
+     *
+     * @param {number} a: Numerator.
+     * @param {number} b: Denominator.
+     * @return {number}: Result of division.
+     */
+    private scalingDivide(a: number, b: number): number {
+        if (b === 0 && a === 0) {
+            return 0;
```

Comment:
I *think* there may be a bug here, and this should be 1. Let me know if I've got a part of this wrong. So if you are holding the origin, and then try to pull (1, 0, 0) to (2, 0, 0), we pass 0 and 0 as `a` and `b` to `scalingDivide` for the y and z axes, and get a scale of 0. If we had a cube before, this would flatten the y and z axes to turn it into a line.

Reply:
I think you're right. When testing with the square there wasn't the problem of flattening the other corners, so returning 0 gave the same result but with a 3D shape it'd be 1.

Reply:
Done